CustomerApp/CustomerApp/Pages/YourOrderPage.xaml.cs

```diff
@@ -60,30 +72,124 @@ async void OnServerButtonClicked(object sender, EventArgs e)
+            if (instructions == item.special_instruct)
+                return;
+
+            OrderItem preUpdateItem = RealmManager.All<Order>().FirstOrDefault().menuItems.Where((OrderItem o) => o.special_instruct == item.special_instruct && o._id == item._id).FirstOrDefault();
+
+            if (preUpdateItem != null)
+            {
+                RealmManager.Write(() => preUpdateItem.special_instruct = instructions);
+
+                await UpdateOrderMenuItemsRequest.SendUpdateOrderMenuItemsRequest(RealmManager.All<Order>().FirstOrDefault()._id, RealmManager.All<Order>().FirstOrDefault().menuItems.ToList());
+            }
+            else
+            {
+                await DisplayAlert("Something went wrong", "Sorry, but we were unable to edit that item's description. Most likely, it was removed by another user at your table. Please verify if the item is gone, then re-add it to your order", "OK");
+            }
+            await DisplayOrder();
         }
 
-        public void DisplayOrder()
+        /// <summary>
+        /// Only called by the Delete swipeView item. Removes the selected item from the order if the order is not yet sent
+        /// Pulls the most recent update for the order, removes the item if it still exists, then updates the remote order.
+        /// Lastly, pulls the most recent order from the server again
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        async void OnRemoveItemInvoked(object sender, EventArgs e)
         {
-            menuFoodItemsView.ItemsSource = RealmManager.All<Order>().FirstOrDefault().Contents.ToList();
+            OrderItem item = new OrderItem((OrderItem)(((SwipeItem)sender).BindingContext));
+
+            await DisplayOrder();
+
+            // Don't allow removal of sent items. Might change this to check if the item is prepared later
+            if (RealmManager.All<Order>().FirstOrDefault().send_to_kitchen)
+            {
+                await DisplayAlert("Option Unavailable", "Sorry, but this option is not available since the order has already been sent", "OK");
+                return;
+            }
+
+            OrderItem preUpdateItem = RealmManager.All<Order>().FirstOrDefault().menuItems.Where((OrderItem o) => o.special_instruct == item.special_instruct && o._id == item._id).FirstOrDefault();
+            
+            // Remove item if it was not already removed
+            if(preUpdateItem != null)
+            {
+                RealmManager.Write(() => RealmManager.All<Order>().FirstOrDefault().menuItems.Remove(preUpdateItem));
+
+                await UpdateOrderMenuItemsRequest.SendUpdateOrderMenuItemsRequest(RealmManager.All<Order>().FirstOrDefault()._id, RealmManager.All<Order>().FirstOrDefault().menuItems.ToList());
+            }
+
+
+            await DisplayOrder();
+        }
+
+        // This and enableRefresh are used in an attempt to make it easier to use the swipeview for each item
```

Comment:
Right now, it is still quite difficult to pull an order left to work with it. it seems to be conflicting with the pull down to refresh. Perhaps the refresh functionality could be placed into a clickable event attached to a refresh button?

This fix seems to work on the first time you try to swipe the order left but every subsequent time after, it becomes pretty challenging.